### ISSUE TITLE
Memory profiler crashes immediately

### ISSUE THREAD

**Issue by jdm** [servo/servo#27836]
Running servo with `-m` on macOS yields:
```
Begin memory reports 5
|
Stack trace for thread "ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }"
   0: <servo::backtrace::Print as core::fmt::Debug>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: servo::backtrace::print
   4: servo::install_crash_handler::handler
   5: __sigtramp

```

**Comment by jdm** [comment]
```
* thread #51, stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x0000000000000000
    frame #1: 0x00000001028c81c1 servo`ShouldMeasureObject(obj=0x000036b144d32040, iface=0x000070000951c1d8) at jsglue.cpp:512:26
    frame #2: 0x0000000102a85ff6 servo`void StatsCellCallback<(Granularity)1>(rt=<unavailable>, data=<unavailable>, cellptr=(ptr = 60134991798336), thingSize=64) at MemoryMetrics.cpp:373:13 [opt]
    frame #3: 0x0000000102ceda38 servo`IterateRealmsArenasCellsUnbarriered(cx=0x000000011c00a000, zone=<unavailable>, data=0x000070000951c3f8, realmCallback=<unavailable>, arenaCallback=(servo`StatsArenaCallback(JSRuntime*, void*, js::gc::Arena*, JS::TraceKind, unsigned long) at MemoryMetrics.cpp:250:61), cellCallback=(servo`void StatsCellCallback<(Granularity)1>(JSRuntime*, void*, JS::GCCellPtr, unsigned long) at MemoryMetrics.cpp:329))(JSContext*, void*, JS::Handle<JS::Realm*>), void (*)(JSRuntime*, void*, js::gc::Arena*, JS::TraceKind, unsigned long), void (*)(JSRuntime*, void*, JS::GCCellPtr, unsigned long)) at PublicIterators.cpp:41:9 [opt]
    frame #4: 0x0000000102ced701 servo`js::IterateHeapUnbarriered(cx=0x000000011c00a000, data=0x000070000951c3f8, zoneCallback=(servo`StatsZoneCallback(JSRuntime*, void*, JS::Zone*) at MemoryMetrics.cpp:204), realmCallback=(servo`StatsRealmCallback(JSContext*, void*, JS::Handle<JS::Realm*>) at MemoryMetrics.cpp:225), arenaCallback=(servo`StatsArenaCallback(JSRuntime*, void*, js::gc::Arena*, JS::TraceKind, unsigned long) at MemoryMetrics.cpp:250:61), cellCallback=(servo`void StatsCellCallback<(Granularity)1>(JSRuntime*, void*, JS::GCCellPtr, unsigned long) at MemoryMetrics.cpp:329))(JSRuntime*, void*, JS::Zone*), void (*)(JSContext*, void*, JS::Handle<JS::Realm*>), void (*)(JSRuntime*, void*, js::gc::Arena*, JS::TraceKind, unsigned long), void (*)(JSRuntime*, void*, JS::GCCellPtr, unsigned long)) at PublicIterators.cpp:57:5 [opt]
    frame #5: 0x0000000102a84cb9 servo`CollectRuntimeStatsHelper(cx=0x000000011c00a000, rtStats=0x000070000951c4b0, opv=0x000070000951c968, anonymize=false, statsCellCallback=(servo`void StatsCellCallback<(Granularity)1>(JSRuntime*, void*, JS::GCCellPtr, unsigned long) at MemoryMetrics.cpp:329))(JSRuntime*, void*, JS::GCCellPtr, unsigned long)) at MemoryMetrics.cpp:643:3 [opt]
    frame #6: 0x0000000102a86893 servo`JS::AddServoSizeOf(cx=<unavailable>, mallocSizeOf=(servo`MallocSizeOf(void const*) at jsglue.cpp:914), opv=<unavailable>, sizes=0x000070000951c9f0)(void const*), JS::ObjectPrivateVisitor*, JS::ServoSizes*) at MemoryMetrics.cpp:850:8 [opt]
    frame #7: 0x00000001028c9af8 servo`::CollectServoSizes(cx=0x000000011c00a000, sizes=0x000070000951c9f0, gs=(servo`script::script_runtime::get_size::h14ab765fdc5815b8 at script_runtime.rs:673)) at jsglue.cpp:935:10
    frame #8: 0x00000001017a7d2f servo`script::script_runtime::get_reports::hbd3fd900b24b00ec(cx=0x000000011c00a000, path_seg=String @ 0x000070000951ced8) at script_runtime.rs:696:12
    frame #9: 0x00000001013d42db servo`script::script_thread::ScriptThread::collect_reports::h076bc6de7dfea83c(self=0x0000700009524fe0, reports_chan=ReportsChan @ 0x000070000951cdac) at script_thread.rs:2542:24
    frame #10: 0x00000001013cb4f6 servo`script::script_thread::ScriptThread::handle_msg_from_script::h401c1dc8cf52ce97(self=0x0000700009524fe0, msg=MainThreadScriptMsg @ 0x000070000951d778) at script_thread.rs:2092:17
    frame #11: 0x00000001013c4015 servo`script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::ha113d0872da72e75 at script_thread.rs:1639:46
    frame #12: 0x00000001013c7a7b servo`script::script_thread::ScriptThread::profile_event::hef2771b659644aea(self=0x0000700009524fe0, category=ScriptEvent, pipeline_id=Option<msg::constellation_msg::PipelineId> @ 0x000070000951da18, f=closure-5 @ 0x00007000095244b0) at script_thread.rs:1882:13
    frame #13: 0x00000001013c21ea servo`script::script_thread::ScriptThread::handle_msgs::hd2e3f2e61e77e19a(self=0x0000700009524fe0) at script_thread.rs:1632:26
    frame #14: 0x00000001013bf9c8 servo`script::script_thread::ScriptThread::start::h9987c7f471c6570b(self=0x0000700009524fe0) at script_thread.rs:1436:15
    frame #15: 0x00000001013b8e2b servo`_$LT$script..script_thread..ScriptThread$u20$as$u20$script_traits..ScriptThreadFactory$GT$::create::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h83215236e599bf21 at script_thread.rs:825:25
    frame #16: 0x0000000101a298f2 servo`profile_traits::mem::ProfilerChan::run_with_memory_reporting::h86296eb806be6bff(self=0x0000700009524fc4, f=closure-1 @ 0x0000700009524e50, reporter_name=String @ 0x00007000095259c8, channel_for_reporter=(flavor = crossbeam_channel::channel::SenderFlavor<script::script_thread::MainThreadScriptMsg> @ 0x0000700009524d80), msg=(0x0000000000000001)) at mem.rs:88:9
    frame #17: 0x00000001013b95c9 servo`_$LT$script..script_thread..ScriptThread$u20$as$u20$script_traits..ScriptThreadFactory$GT$::create::_$u7b$$u7b$closure$u7d$$u7d$::ha94ae7d333c24743 at script_thread.rs:823:17
    frame #18: 0x0000000100b0f143 servo`std::sys_common::backtrace::__rust_begin_short_backtrace::hd22f034984541edd(f=<unavailable>) at backtrace.rs:130:5
    frame #19: 0x000000010146c003 servo`std::thread::Builder::spawn_unchecked::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h052bcc17fed80cbc at mod.rs:475:17
    frame #20: 0x00000001022613f3 servo`_$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$$LP$$RP$$GT$$GT$::call_once::h765fb645bdf4e64c(self=<unavailable>, _args=<unavailable>) at panic.rs:318:9
    frame #21: 0x000000010146d438 servo`std::panicking::try::do_call::hdeafaef47bef5c91(data="��R+\x01") at panicking.rs:348:40
    frame #22: 0x000000010168fd9d servo`__rust_try + 29
    frame #23: 0x000000010146d077 servo`std::panicking::try::heb18c80f46a8fe23(f=<unavailable>) at panicking.rs:325:15
    frame #24: 0x0000000102277f33 servo`std::panic::catch_unwind::he43d26b62c29ad06(f=<unavailable>) at panic.rs:394:14
    frame #25: 0x000000010146b0a0 servo`std::thread::Builder::spawn_unchecked::_$u7b$$u7b$closure$u7d$$u7d$::h0dc5fd5a08fb4f6f at mod.rs:474:30
    frame #26: 0x0000000101a55521 servo`core::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h63f8a5f0d39ebefc((null)=0x000000012b6b6000, (null)=<unavailable>) at function.rs:233:5
    frame #27: 0x0000000106b8691d servo`std::sys::unix::thread::Thread::new::thread_start::h834ded9490287f71 [inlined] _$LT$alloc..boxed..Box$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once::h4d3d821adf25cdf6 at boxed.rs:1074:9 [opt]
    frame #28: 0x0000000106b86917 servo`std::sys::unix::thread::Thread::new::thread_start::h834ded9490287f71 [inlined] _$LT$alloc..boxed..Box$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$A$GT$$GT$::call_once::h6f972c70872e9c5d at boxed.rs:1074 [opt]
    frame #29: 0x0000000106b8690e servo`std::sys::unix::thread::Thread::new::thread_start::h834ded9490287f71 at thread.rs:87 [opt]
    frame #30: 0x00007fff73c76109 libsystem_pthread.dylib`_pthread_start + 148
    frame #31: 0x00007fff73c71b8b libsystem_pthread.dylib`thread_start + 15
```

**Comment by jdm** [comment]
Ah, we haven't set a callback function to decide whether to measure a JS object or not by calling https://github.com/servo/rust-mozjs/blob/04fd4b7066f1b1bac2de4c8ccc53a2247dcd7e86/src/jsglue.cpp#L939.

**Comment by jdm** [comment]
This looks like an accidental regression from https://github.com/servo/servo/commit/74c1e00d8163f255bb4141ff3549bbdedd7ea766#diff-86033f852232a3ea1cb2f43d2f255850fb6d091c6eeba07213152e13b7006026L219.

**Comment by jdm** [comment]
We should be able to resurrect that code in https://github.com/servo/servo/blob/20ef49e035596136ce43b47a66b27c8b5c25967b/components/script/init.rs#L53.

**Comment by joseph62** [comment]
Hello! I would like to work on this issue. 

I started looking into restoring the `is_dom_object` and hit a snag. `is_platform_object` has changed since this was removed (commit [63714c90fb](https://github.com/servo/servo/commit/63714c90fb5bbad86f28fc188120b2ecfd3337ab#diff-8c5d599e1ef8986fee0e2124772cfa9fd8ae06f617368280cd08a183096c8300R213)) and now requires a JSContext as a second argument. How might I account for this change? 

@highfive: assign me
